src/System.Runtime.Extensions/tests/Performance/Perf.StreamWriter.cs

```diff
@@ -0,0 +1,119 @@
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [MemberData(nameof(WriteMemberData))]
+        public void WritePartialCharArray(int writeLength)
+        {
+            char[] buffer = new string('a', writeLength + 10).ToCharArray();
+            int innerIterations = MemoryStreamSize / writeLength;
+            int outerIteration = TotalWriteCount / innerIterations;
+            using (var stream = new MemoryStream(MemoryStreamSize))
+            {
+                using (var writer = new StreamWriter(stream, new UTF8Encoding(false, true), DefaultStreamWriterBufferSize, true))
+                {
+                    foreach (var iteration in Benchmark.Iterations)
+                    {
+                        using (iteration.StartMeasurement())
+                        {
+                            for (int i = 0; i < outerIteration; i++)
+                            {
+                                for (int j = 0; j < innerIterations; j++)
+                                {
+                                    writer.Write(buffer, 10, writeLength);
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [MemberData(nameof(WriteMemberData))]
+        public void WriteString(int writeLength)
+        {
+            string value = new string('a', writeLength);
+            int innerIterations = MemoryStreamSize / writeLength;
+            int outerIteration = TotalWriteCount / innerIterations;
+            using (var stream = new MemoryStream(MemoryStreamSize))
+            {
+                using (var writer = new StreamWriter(stream, new UTF8Encoding(false, true), DefaultStreamWriterBufferSize, true))
+                {
+                    foreach (var iteration in Benchmark.Iterations)
+                    {
+                        using (iteration.StartMeasurement())
+                        {
+                            for (int i = 0; i < outerIteration; i++)
+                            {
+                                for (int j = 0; j < innerIterations; j++)
+                                {
+                                    writer.Write(value);
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> WriteMemberData()
+        {
+            for (int i = 2; i <= 10; i++)
```

Comment:
My understanding for the existence of performance tests is to catch regressions in the future. I chose the values I did as small strings/char arrays are very important for serialization so are important scenarios. If in the future someone chooses to replace this change with improvements to Buffer.BlockCopy as @stephentoub [suggested](https://github.com/dotnet/corefx/pull/19126#issuecomment-298228450) then it's important to see that it improves things across this entire range. I can definitely see justification to reduce it to 2 -> 6. As there is a tiny regression caused by the extra if check when taking the original code path, I included the 100 value to show that the regression doesn't scale with input size. Although my intention was to show it doesn't regress in that case, seeing that there are zero other perf tests for this class there might still be value. Although you could argue that it's mainly measuring the perf for Buffer.BlockCopy, the write methods do also handle partial write to an internal buffer, flushing, then writing the remainder so it is doing something subtly different.  
Ultimately I don't own this part of corefx so I'll modify the test values to whatever the stakeholders feel is best.